packages/perseus/src/widgets/sorter/sorter-validator.test.ts

```diff
@@ -0,0 +1,41 @@
+import sorterValidator from "./sorter-validator";
+
+import type {Rubric, UserInput} from "./sorter.types";
+
+describe("SorterValidator", () => {
```

Comment:
Looking at that code, it depends in the React components `state`.  

What we could do is include `changed` in the `UserInput` and then move this logic from `simpleValidate` to the new static validator. 

What do you think about that approach?

Reply:
Sounds great! Was able to add it to UserInput and refactor code to include it. Also was able to move over the invalid logic and add a related test! Thanks~ :D

Reply:
@handeyeco mentioned changing the name of `options` to `values` in `getUserInput` might be bad since this is an external API and we don't know how it's used. Could modifying the return object of `getUserInput` be similarly bad? If I'm just adding something that no other code is currently using, it should be fine, right?

Reply:
I think you're safe if you're adding a field. There _may_ be call sites that call `Sorter.validate(...)` in webapp, but we'll need to visit those sites when we full remove `static validate` from our widgets so we'll need to ensure that a `changed` flag is passed through properly for this widget.

Reply:
Yeah, I'm trying to be cautious about changing the external API of things, but in this case I think y'all made the right decision.